packages/preferences/src/browser/abstract-resource-preference-provider.ts

```diff
@@ -26,7 +26,7 @@ export abstract class AbstractResourcePreferenceProvider extends PreferenceProvi
     protected async init(): Promise<void> {
         const uri = await this.getUri();
         this.resource = this.resourceProvider(uri);
-        this.readPreferences();
+        this.readPreferences().then(() => this._ready.resolve());
```

Comment:
What if it fails? Should not we await?

Reply:
I don't think readPreferences can fail currently (nothing in it can throw), but it would still be a good safety net to resovle _ready on failure.  Otherwise, it could hang forever because the preference service will never become ready.  I'll change it to

```
        this.readPreferences()
            .then(() => this._ready.resolve())
            .catch(() => this._ready.resolve());
```